File: config-biz/src/main/java/io/gitee/zicai/biz/service/AppService.java
package io.gitee.zicai.biz.service;

import io.gitee.zicai.core.entity.App;
import io.gitee.zicai.biz.mapper.AppMapper;
import org.springframework.stereotype.Service;
import tk.mybatis.mapper.weekend.Weekend;

@Service
public class AppService extends BaseService<AppMapper, App> {

    public App findOne(String appName, String appEnv) {
        Weekend<App> weekend = new Weekend<>(App.class);
        weekend.weekendCriteria().andEqualTo(App::getAppName, appName).andEqualTo(App::getAppEnv, appEnv);

        return mapper.selectOneByExample(weekend);
    }
}

File: config-biz/src/main/java/io/gitee/zicai/biz/service/AppService.kt
package io.gitee.zicai.biz.service

import com.baomidou.mybatisplus.extension.kotlin.KtQueryWrapper
import io.gitee.zicai.biz.mapper.AppMapper
import io.gitee.zicai.core.entity.App
import org.springframework.stereotype.Service

@Service
open class AppService : BaseService<AppMapper, App>() {

    fun findOne(appName: String, appEnv: String): App {
        val wrapper = KtQueryWrapper(App())
        wrapper.eq(App::appName, appName).eq(App::appEnv, appEnv)

        return baseMapper.selectOne(wrapper)
    }
}
